recipe_manager/serializers.py

```diff
@@ -0,0 +1,45 @@
+# serializers.py
+from rest_framework import serializers
+from .models import Recipe, Ingredient
+
+class IngredientSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Ingredient
+        fields = ['id', 'name']
+
+class RecipeSerializer(serializers.ModelSerializer):
+    ingredients = IngredientSerializer(many=True)
+
+    class Meta:
+        model = Recipe
+        fields = ['id', 'recipe_name', 'instructions', 'ingredients']
+
+    def create(self, validated_data):
+        ingredients_data = validated_data.pop('ingredients')
+        recipe = Recipe.objects.create(**validated_data)
+        for ingredient_data in ingredients_data:
+            Ingredient.objects.create(recipe=recipe, **ingredient_data)
+        return recipe
```

Comment:
There's one interesting thing here, I don't think you need to change it for the purpose of this exercise, but I thought it might be worth mentioning.

If for whatever reason an ingredient fails to be inserted (e.g. setting by setting the recipe_id to a non-existing recipe like this `Ingredient.objects.create(recipe_id=1337, **ingredient_data)`), the database will respond with an integrity error because this recipe doesn't exist, however the previous database operations until this point will still succeed. 

So you end up reporting an http error to the user, but some parts might have been committed to the database actually. You can change that behaviour by either enabling atomic requests (which is what our Django backend has, so each http request is wrapped in an atomic transaction), or you can wrap this block in an atomic block. This basically means that all of the transaction needs to succeed in order to be committed, and not only parts

More can be read here in case you are interested https://docs.djangoproject.com/en/5.0/topics/db/transactions/ (again, for the exercise it's fine to have it this way, but just that you are aware 😄)

Reply:
makes sense! thanks for that